flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointTestBase.java

```diff
@@ -224,8 +224,8 @@ public InputStatus pollNext(ReaderOutput<Long> output) {
 
 			@Override
 			public void notifyCheckpointComplete(long checkpointId) {
-				LOG.info("notifyCheckpointComplete {} @ {} subtask (? attempt)", split.numCompletedCheckpoints, split.nextNumber % split.increment);
 				if (split != null) {
+					LOG.info("notifyCheckpointComplete {} @ {} subtask (? attempt)", split.numCompletedCheckpoints, split.nextNumber % split.increment);
```

Comment:
Can `split.increment` be 0 here?

Reply:
`Increment = parallelism`, so it would be a bug at a different point. But good question.